incubator/hnc/internal/validators/object.go

```diff
@@ -108,6 +109,10 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 
 	// If the object wasn't and isn't inherited, it's none of our business.
 	if !oldInherited && !newInherited {
+		if yes, cnses := o.hasConflict(inst); yes {
+			msg := fmt.Sprintf("Cannot create and propagate this object because one or more objects with the same name already exist in the descendants: %v. Please resolve the conflict first.", cnses)
```

Comment:
"This object" might be ambiguous, e.g. if the kubectl command is running in a script and this log message is all the user has to go on, or if the user didn't specify the namespace and is using the one in their kubectx. Maybe include a bunch more information, e.g.:

```
Cannot create <gvk> <name> in namespace <ns> because it would overwrite objects in the following descendant namespace(s):
  <ns 1>
  <ns 2>
Choose a different name for the object, or remove the conflicting objects from these namespaces.
```